src/index.ts

```diff
@@ -32,26 +32,78 @@ async function run(): Promise<void> {
     const titleRegexFlags = getInput('titleRegexFlags', {
       required: true
     });
+    const ticketLink = getInput('ticketLink', { required: false });
     const titleRegex = new RegExp(titleRegexBase, titleRegexFlags);
-    const titleCheck = title.match(titleRegex);
+    const titleCheck = titleRegex.exec(title);
+
+    // Instantiate a GitHub Client instance
+    const token = getInput('token', { required: true });
+    const client = getOctokit(token);
+    const { owner, repo, number } = context.issue;
+    const login = context.payload.pull_request?.user.login as string;
+    const senderType = context.payload.pull_request?.user.type as string;
+    const sender: string = senderType === 'Bot' ? login.replace('[bot]', '') : login;
+
+    const linkTicket = async (matchArray: RegExpMatchArray): Promise<void> => {
+      debug('match array for linkTicket', JSON.stringify(matchArray));
+      debug('match array groups for linkTicket', JSON.stringify(matchArray.groups));
+
+      if (!ticketLink) {
+        return;
+      }
+
+      const ticketNumber = matchArray.groups?.ticketNumber;
+
+      if (!ticketNumber) {
+        debug('ticketNumber not found', 'ticketNumber group not found in match array.');
+
+        return;
+      }
+
+      if (!ticketLink.includes('%ticketNumber%')) {
+        debug('invalid ticketLink', 'ticketLink must include "%ticketNumber%" variable to post ticket link.');
+
+        return;
+      }
+
+      const linkToTicket = ticketLink.replace('%ticketNumber%', ticketNumber);
+
+      const currentReviews = await client.pulls.listReviews({
+        owner,
+        repo,
+        pull_number: number
+      });
+
+      debug('current reviews', JSON.stringify(currentReviews));
+
+      if (
+        currentReviews?.data?.length &&
+        currentReviews?.data.some((review: { body?: string }) => review?.body?.includes(linkToTicket))
+      ) {
+        debug('already posted ticketLink', 'found an existing review that contains the ticket link');
+
+        return;
+      }
```

Comment:
Added some logic here to prevent needlessly creating duplicate ticket link comments when the action runs again due to e.g. updating the PR body.
It still creates another ticket link comment if the ticket number detected changes (although I didn't handle deleting the old comment in that scenario).